### PR TITLE
VACMS-13479: Applies D10-compliant minor changes to custom modules

### PR DESCRIPTION
**Pull request by tonytaylor**
## Description

Applies minor changes for D10 compliance to the following custom modules:

- va_gov_api
- va_gov_backend
- va_gov_db
- va_gov_events
- va_gov_facilities
- va_gov_migrate
- va_gov_notifications
- va_gov_post_api
- va_gov_resources_and_support
- va_gov_vamc
- va_gov_vet_center

Relates to #13479 

## Testing done

- Verified no WSOD
- Verified most forms properly render post-refactoring

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

As user _uid_ with _user_role_
1. Do this
   - [ ] Validate that
2. Then
   - [ ] Validate that
3. Then validate Acceptance Criteria from issue
   - [ ] a
   - [ ] b
   - [ ] c

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteCmsLinksValidator.php
Comment (ndouglas) [inline docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteCmsLinksValidator.php:1]: Why is this refactor (factoring out common code into TextValidatorTrait) being reversed?

File: docroot/modules/custom/va_gov_backend/src/Service/VaGovUrl.php [hunk 1]
```diff
@@ -65,7 +64,7 @@ public function getVaGovFrontEndUrlForEntity(EntityInterface $entity) : string {
     try {
       return $this->getVaGovFrontEndUrl() . $entity->toUrl()->toString();
     }
-    catch (Exception $e) {
+    catch (\Exception $e) {
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_backend/src/Service/VaGovUrl.php:67]: 👍🏻

File: docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php [hunk 2]
```diff
@@ -85,7 +85,7 @@ public function checkForOutdatedVamcContent(): array {
           $this->vaGovNotifications
             ->info('Outdated content found for @sectionName editor: @editor',
             ['@editor' => $editorName, '@sectionName' => $sectionName]);
-          $this->queueNotification($editor, 'vamc_outdated_content', 'vamc_outdated_content');
+          $this->queueNotification($editor, 'vamc_outdated_content');
```
Comment (ndouglas) [inline docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php:88]: Doesn't this undo some of @edmund-dunn's work?

**Review by ndouglas — COMMENTED** [review]
Mostly solid, but I feel like some merges might've gone the wrong way.

**Comment by tonytaylor** [comment]
Thanks @ndouglas.  I guess there's some confusion surrounding this.

File: docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php (hunk 2 above)
Comment (tonytaylor) [reply docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php:88]: That is @edmund-dunn's work.
I diffed this against the D10 branch to verify.

File: docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteCmsLinksValidator.php
Comment (tonytaylor) [reply docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteCmsLinksValidator.php:1]: These are from the D10 branch.

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php [hunk 3]
```diff
@@ -192,29 +193,6 @@ protected function getPayload(bool $forcePush = FALSE): array {
...
-      // let's do our best to trim this down if we need to
-      if ($operatingStatusMoreInfoLength > 300) {
-        $operatingStatusMoreInfo = str_replace('&nbsp;', ' ', $operatingStatusMoreInfo);
-        $operatingStatusMoreInfo = trim($operatingStatusMoreInfo);
-        $operatingStatusMoreInfo = preg_replace("/(\r?\n|\r)+/", " ", $operatingStatusMoreInfo);
-      }
-    }
-    return $operatingStatusMoreInfo;
-  }
-
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php:217]: This is relatively new code that should not be going away.

File: docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php [hunk 4]
```diff
@@ -169,7 +170,7 @@ protected function getPayload(bool $forcePush = FALSE): array {
 
     // Current field values.
     $this->statusToPush = $this->facilityNode->get('field_operating_status_facility')->value;
-    $this->additionalInfoToPush = $this->getOperatingStatusMoreInfoShort();
+    $this->additionalInfoToPush = $this->facilityNode->get('field_operating_status_more_info')->value;
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php:173]: This change should not be happening.

**Comment by tonytaylor** [comment]
Thanks guys.  I appreciate the feedback.

I think I've cleaned this up.

File: docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php (hunk 2 above)
Comment (ndouglas) [reply docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php:88]: That might be your issue -- I don't know if that work is present in the D10 branch.

**Review by ndouglas — CHANGES_REQUESTED** [review]
Everything looks wonderful _except_:

Changes to these files appear revert some work that I think was done after the D10 branch was split from `main`:

- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteCmsLinksValidator.php`
- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteUrlsAsPathsInLinksValidator.php`
- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventDomainsAsPathsInLinksValidator.php`
- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventProtocolRelativeLinksValidator.php`
- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventVaGovDomainsAsPathsInLinksValidator.php`
- `docroot/modules/custom/va_gov_notifications/src/Service/OutdatedContent.php`
- `docroot/modules/custom/va_gov_notifications/templates/message--va-outdated-content--mail-body.html.twig`
- `docroot/modules/custom/va_gov_notifications/va_gov_notifications.module`
- `docroot/modules/custom/va_gov_post_api/src/Service/PostFacilityStatus.php`

**Comment by tonytaylor** [comment]
This was a rather unproductive rabbit hole. Closing.

**Pull request closed without merging**